lunar-client: adapt and update to 3.0.7

surfaceflinger opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Bumped the version to 3.0.7.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

surfaceflinger commented:
this update changed icon name from lunarclient.png to a very generic launcher.png, not sure if this will conflict with anything but I left it as it is

LostAttractor reviewed (approved):
The icon name change didn't cause any problem on my system, it's not clear if there is something else

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2459

SuperSandro2000 commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -2,12 +2,12 @@
     name = "lunar-client";
     exec = "lunar-client";
-    icon = "lunarclient";
+    icon = "launcher";
```
Isn't that name super generic and could easily clash?

surfaceflinger replied:
Should be better now :)

surfaceflinger commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -30,7 +30,10 @@ appimageTools.wrapType1 rec {
   extraInstallCommands = ''
     mkdir -p $out/share/applications
     cp ${desktopItem}/share/applications/* $out/share/applications
```
How could this be done with appimageTools wrap?

surfaceflinger committed:
lunar-client: adapt and update to 3.0.7

surfaceflinger commented:
Lunar Client 3.0.5 got released and basically changed many things. I adapted and updated the derivation, also added myself as a maintainer. Not sure if this can be in a single commit like this.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2582

SuperSandro2000 commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -1,45 +1,37 @@
-  };
+  pname = "lunar-client";
+  version = "3.0.6";
+  name = "${pname}-${version}";
```
We should drop name as it isn't really required anymore

SuperSandro2000 commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -1,45 +1,37 @@
+    hash = "sha256-ujDiGfs2py/Sanh1sgXGLfP4ergm3O4p90B4M3WgU54=";
   };
+
+  appimageContents = appimageTools.extract { inherit name src; };
```
```suggestion
  appimageContents = appimageTools.extract { inherit pname version src; };
```

SuperSandro2000 commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -1,45 +1,37 @@
-    mkdir -p $out/share/applications
-    cp ${desktopItem}/share/applications/* $out/share/applications
-    cp -r ${appimageContents}/usr/share/icons/ $out/share/
+    mv $out/bin/{${name},${pname}}
```
```suggestion
    mv $out/bin/{${pname}-${version},${pname}}
```

SuperSandro2000 commented on pkgs/games/lunar-client/default.nix:
```diff
@@ -1,45 +1,37 @@
-    cp ${desktopItem}/share/applications/* $out/share/applications
-    cp -r ${appimageContents}/usr/share/icons/ $out/share/
+    mv $out/bin/{${name},${pname}}
+    install -Dm444 ${appimageContents}/launcher.desktop $out/share/applications/${pname}.desktop
+    install -Dm444 ${appimageContents}/launcher.png $out/share/pixmaps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}' \
+      --replace 'Icon=launcher' 'Icon=${pname}'
```
We shouldn't be using pname here so often

surfaceflinger replied:
Should I just write lunar-client here explicitly?

SuperSandro2000 replied:
yeah

surfaceflinger replied:
done!
